src/lib/codec-importer.ts

```diff
@@ -0,0 +1,47 @@
+import multicodecs from 'multicodec'
+import { BlockCodec } from 'multiformats/codecs/interface'
+import type { PBNode } from '@ipld/dag-pb'
+import type { IPLDFormat } from 'blockcodec-to-ipld-format'
+
+export default async function codecImporter<T extends any = unknown>(codecCode: number): Promise<BlockCodec<any, any> | IPLDFormat<any>> {
```

Comment:
I *really* don't like that there isn't a common library for doing this. 

1. multicodec is deprecated - BUT finding the codes is *not* easy no matter how many times I view the multicodec table. 
2. no current library contains all codecs (makes sense)
3. Hardcoding magic numbers here feels wrong. 
4. Not all codecs are using blockCodec format, and therefore don't export a "code" property. 

We really need a solution to dynamically import codecs (from IPFS!?) based on codec code. A library with tree-shaking, code-splitting, and dynamic import support could really go a long way to helping the js ecosystem with ipld stuffs.